_field-types/flat-object.md

```diff
@@ -0,0 +1,221 @@
+---
+layout: default
+title: Flat object
+nav_order: 43
+has_children: false
+parent: Object field types
+grand_parent: Supported field types
+---
+
+# Flat object field type
+
+In OpenSearch, you don't have to specify a mapping before indexing documents. If you don't specify a mapping, OpenSearch uses [dynamic mapping]({{site.url}}{{site.baseurl}}/field-types/mappings#dynamic-mapping) to map every field and its subfields in the document automatically. When you ingest documents such as logs, you may not know every field's subfield name and type in advance. In this case, dynamically mapping all new subfields can quickly lead to a "mapping explosion", where the growing number of fields may degrade the performance of your cluster. 
+
+Flat object solves this problem by not indexing the field itself or its subfields. Instead, for any JSON object, the values of all subfields and their paths are stored in two string fields. Subfields within the JSON are accessible using standard dot path notation but are not indexed for fast lookup.
+
+The maximum field path length in the dot notation is 2<sup>24</sup> &minus; 1.
+{: .note}
+
+Flat objects provide the following advantages:
+
+- Efficient reads: Fetching performance is similar to that of a keyword field.
+- Memory efficiency: Storing the entire complex JSON object in one field without indexing all its subfields reduces the number of fields in an index. 
+- Space efficiency: OpenSearch does not create an inverted index for flat objects, thereby saving space. 
+- Compatibility for migration: You can migrate your data from systems that support similar flat types to OpenSearch.
+
+Mapping a field as flat object applies when a field and its subfields are mostly read and not used as a search criteria because the subfields are not indexed. Flat objects are useful for objects with a large number of fields or when you don't know the keys in advance.
+
+Flat objects support:
+
+- Exact match queries. 
+- Textual sorting.
```

Comment:
Not supported for sorting, aggregation and filtering

Reply:
@mingshl So both textual  and numerical sorting are not supported?